src/Rules/Generate.hs

```diff
@@ -85,15 +85,18 @@ compilerDependencies stage =
        , "primop-vector-tys.hs-incl"
        , "primop-vector-uniques.hs-incl" ]
 
--- TODO: Turn this into a FilePaths expression
-generatedDependencies :: Stage -> Package -> [FilePath]
-generatedDependencies stage pkg
-    | pkg   == compiler = compilerDependencies stage
-    | pkg   == ghcPrim  = ghcPrimDependencies stage
-    | pkg   == rts      = libffiDependencies ++ includesDependencies
-                       ++ derivedConstantsDependencies
-    | stage == Stage0   = includesDependencies
-    | otherwise         = []
+generatedDependencies :: Expr [FilePath]
+generatedDependencies = do
+    pkg <- getPackage
+    stage <- getStage
+    let deps
+            | pkg   == compiler = compilerDependencies stage
+            | pkg   == ghcPrim  = ghcPrimDependencies stage
+            | pkg   == rts      = libffiDependencies ++ includesDependencies
+                               ++ derivedConstantsDependencies
+            | stage == Stage0   = includesDependencies
+            | otherwise         = []
+    return deps
 
```

Comment:
I was actually thinking to use predicates, e.g.:
```haskell
generatedDependencies = mconcat
   [ package compiler ? compilerDependencies
   , ...
   , stage0           ? includeDependencies ]
```
And make `compilerDependencies` and `includeDependencies` expressions too.